python/amici/petab_objective.py

```diff
@@ -155,6 +159,114 @@ def simulate_petab(
+    :sep:
+        Separator for csv file for export
+    :return:
+        Pandas Dataframe containing
+        * the names of the state variables as row indices,
+        * the indices of the amici edata objects as columns,
+          (**NOTE**: only edatas with timepoints INF will be considered)
+        * the steady state values of the state variables as table content`.
+    """
+
+    # if steady state measurements are available: get the simulation conditions
+    steadystate_conditions = _find_steadystate_conditions(petab_problem)
+
+    if simulation_conditions is not None:
+        # filter simulation conditions for steadystate simulations
+        simulation_conditions = _filter_simulation_conditions(
+            simulation_conditions, steadystate_conditions)
+    else:
+        simulation_conditions = steadystate_conditions
+    # get a list identifying the conditions to be simulated
+    condition_list = _create_condition_list(simulation_conditions)
+
+    if edatas is not None:
+        # edatas overwrites simulation_conditions: Overwrite condition_list.
+        condition_list = [f'edata_{ie}' for ie, edata in enumerate(edatas)
+                          if any(np.isinf(edata.getTimepoints()))]
+        # filter edata objects for steadystate simulations
+        edatas = [edata for edata in edatas
+                  if any(np.isinf(edata.getTimepoints()))]
+
+    if not condition_list:
+        if export_file is not None:
+            logging.warning('No output file could be written, as no matching '
+                            'preequilibration and simulation condition with '
+                            'steady state data was found.')
+        # no steady state conditions found. Return an empty dataframe.
+        return pd.DataFrame(columns=amici_model.getStateIds(),
+                            index=[], data=[])
+
+    # simulate the model
+    sim_results = simulate_petab(petab_problem=petab_problem,
+                                 amici_model=amici_model,
+                                 solver=solver,
+                                 problem_parameters=problem_parameters,
+                                 simulation_conditions=simulation_conditions,
+                                 edatas=edatas,
+                                 parameter_mapping=parameter_mapping,
+                                 scaled_parameters=scaled_parameters,
+                                 log_level=log_level,
+                                 num_threads=num_threads)
+    # get the steady states (postequilibration only, use last timepoint)
+    steady_states = np.array([rdata['x'][-1,:]
+                              for rdata in sim_results['rdatas']])
+    # create dataframe
+    steady_state_df = pd.DataFrame(columns=amici_model.getStateIds(),
+        index=condition_list, data=steady_states)
+
+    # export to file (if requested)
+    if export_file is not None:
+        steady_state_df.to_csv(export_file, sep=sep, index=False)
```

Comment:
I'd leave that out of this function. Adds unnecessary complexity to function body and signature. Saving it outside is not that big a deal.